src/agentlab/agents/generic_agent/generic_agent.py

```
@@ -74,10 +74,15 @@ def obs_preprocessor(self, obs: dict) -> dict:
     @cost_tracker_decorator
     def get_action(self, obs):
 
+        goal_object = obs.pop("goal_object", [])
+        if len(goal_object) == 0:
```

just remove [] and let the error raise itself